app/src/main/java/ru/vssemikoz/newsfeed/storage/NewsStorage.java

```diff
@@ -16,25 +15,19 @@ public NewsStorage(NewsItemDAO newsItemDAO) {
         this.newsItemDAO = newsItemDAO;
     }
 
-    public List<NewsItem> getNewsFromDB(boolean favoriteNewsState, Category category, Source source) {
-        if (favoriteNewsState & category == Category.ALL & source == Source.ALL) {
+    public List<NewsItem> getNewsFromDB(boolean favoriteNewsState, Category category) {
+        if (favoriteNewsState & category == Category.ALL) {
```

Comment:
This doesn't work as expected since logical or looks like this `&&` but not like `&`. BTW, all methods checks don't look optimal. We should try to make them simpler.